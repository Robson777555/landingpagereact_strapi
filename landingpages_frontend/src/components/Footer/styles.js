import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../../components/TextComponent/styles';
import { Container as SectionContainer } from '../SectionContainer/styles';

export const Container = styled.footer`
  /* eslint-disable-next-line no-unused-vars */
  ${({ theme }) => css`
    font-family:${theme.fonts.family.default};
    text-align: center;
    font-size: ${theme.fonts.sizes.small};
    border-top: 0.1rem solid ${theme.colors.mediumGray};
    
    a{
      color: inherit;
      text-decoration: none;
      
    }
    & ${TextComponent} {
      font-size: ${theme.fonts.sizes.small};
    }

    & ${SectionContainer} {
      padding-top: 0;
      padding-bottom: 0;
    }
  `}
  
`;
