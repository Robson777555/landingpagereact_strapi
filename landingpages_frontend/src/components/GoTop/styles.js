import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    bottom: 2rem;
    right: 2rem;
    border-radius: 50%;
    z-index: 6;
    opacity: 0.7;
    transition: all 0.3s ease;
    cursor: pointer;

    /* Melhorias para mobile */
    @media (max-width: 768px) {
      width: 3.5rem;
      height: 3.5rem;
      bottom: 1.5rem;
      right: 1.5rem;
    }

    &:hover {
      opacity: 1;
      transform: scale(1.05);
    }

    /* Garante que seja clic√°vel em mobile */
    -webkit-tap-highlight-color: transparent;
    touch-action: manipulation;
  `}
`;
