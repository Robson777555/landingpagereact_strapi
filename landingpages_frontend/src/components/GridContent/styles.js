import styled, { css } from 'styled-components';

export const Container = styled.div`
  /* eslint-disable-next-line no-unused-vars */
  ${({ theme }) => css`
    font-family:${theme.fonts.family.default}; 
    text-align: center;
    max-width: 58rem;
    margin: 0 auto;
    

  `}
`;

export const html = styled.div`
  /* eslint-disable-next-line no-unused-vars */
  ${({ theme }) => css`
    font-family:${theme.fonts.family.default}; 
    margin:${theme.spacings.xhuge} 0;
    text-align: center;
    
    pre {
      display: inline-block;
      
      margin: 0 auto;
    }

  `}
`;
