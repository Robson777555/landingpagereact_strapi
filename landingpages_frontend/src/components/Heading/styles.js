import styled, { css } from 'styled-components';

const titleSize = {
  small: (theme) => css`
    font-size: ${theme.fonts.sizes.medium};
  `,
  medium: (theme) => css`
    font-size: ${theme.fonts.sizes.large};
  `,
  big: (theme) => css`
    font-size: ${theme.fonts.sizes.xlarge};
  `,
  huge: (theme) => css`
    font-size: ${theme.fonts.sizes.xhuge};
    ${css`
      @media (max-width: 768px) {
        font-size: ${theme.fonts.sizes.xlarge};
      }
    `}
  `,
};

const titleCase = (uppercase) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1`
  ${({ theme, $colorDark, $size, $uppercase }) => css`
    color: ${$colorDark ? theme.colors.primaryColor : theme.colors.white};
    background: transparent;
    padding: 1rem;
    
    ${titleSize[$size](theme)};
    ${titleCase($uppercase)};
  `}
`;
