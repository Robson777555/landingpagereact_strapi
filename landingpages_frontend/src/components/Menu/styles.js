import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme, $isVisible }) => css`
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    z-index: 100;
    background: ${theme.colors.white};
    box-shadow: 0 2px 5px rgba(0,0,0,0.1);
    transition: all 0.3s ease;

    /* Desktop: remove comportamento fixo */
    @media (min-width: 769px) {
      position: static;
      height: auto !important;
      overflow: visible !important;
      box-shadow: none;
    }

    /* Mobile: mantÃ©m comportamento de menu */
    @media ${theme.media.lteMedium} {
      height: ${$isVisible ? '100vh' : '60px'};
      overflow: ${$isVisible ? 'auto' : 'hidden'};
    }
  `}
`;

export const MenuContent = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    padding: ${theme.spacings.medium} 0;

    @media (min-width: 769px) {
      flex-direction: row;
      align-items: center;
      justify-content: space-between;
      padding: ${theme.spacings.small} 0;
    }
  `}
`;

export const Button = styled.button`
  ${({ theme }) => css`
    display: none;
    position: fixed;
    top: ${theme.spacings.small};
    right: ${theme.spacings.small};
    z-index: 101;
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    border: none;
    width: 4rem;
    height: 4rem;
    border-radius: 50%;
    cursor: pointer;

    @media ${theme.media.lteMedium} {
      display: flex;
      align-items: center;
      justify-content: center;
    }
  `}
`;
