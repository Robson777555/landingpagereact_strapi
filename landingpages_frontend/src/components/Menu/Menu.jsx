import { useState } from 'react';
import P from 'prop-types';
import * as Styled from './styles';
import { SectionContainer } from '../SectionContainer/SectionContainer';
import { LogoLink } from '../LogoLink/LogoLink';
import { NavLinks } from '../NavLinks/NavLinks';
import { Menu as MenuIcon } from '@styled-icons/material-outlined/Menu';
import { Close as CloseIcon } from '@styled-icons/material-outlined/Close';
import { useEffect } from 'react';

export const Menu = ({ links = [], logoData }) => {
  const [isVisible, setIsVisible] = useState(false);
  const [isMobile, setIsMobile] = useState(false);

  useEffect(() => {
    const checkMobile = () => {
      setIsMobile(window.innerWidth <= 768);
    };

    checkMobile();
    window.addEventListener('resize', checkMobile);
    return () => window.removeEventListener('resize', checkMobile);
  }, []);

  const handleLinkClick = () => {
    if (isMobile) {
      setIsVisible(false);
    }
  };

  return (
    <>
      <Styled.Container $isVisible={isVisible}>
        <SectionContainer>
          <Styled.MenuContent>
            <LogoLink {...logoData} />
            <NavLinks links={links} onLinkClick={handleLinkClick} />
          </Styled.MenuContent>
        </SectionContainer>
      </Styled.Container>

      {isMobile && (
        <Styled.Button
          onClick={() => setIsVisible(!isVisible)}
          aria-label={isVisible ? 'Fechar menu' : 'Abrir menu'}
        >
          {isVisible ? <CloseIcon /> : <MenuIcon />}
        </Styled.Button>
      )}
    </>
  );
};
Menu.propTypes = {
  links: P.arrayOf(
    P.shape({
      children: P.string.isRequired,
      link: P.string.isRequired,
      newTab: P.bool,
    }),
  ),
  logoData: P.shape({
    text: P.string.isRequired,
    link: P.string.isRequired,
  }).isRequired,
};

export default Menu;
