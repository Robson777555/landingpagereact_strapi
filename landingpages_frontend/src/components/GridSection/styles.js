import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { Container as TextComponent } from '../TextComponent/styles';
export const Container = styled.div`
  /* eslint-disable-next-line no-unused-vars */
  ${({ theme }) => css`
   font-family: ${theme.fonts.family.default};
   ${TextComponent} {
    margin-bottom: ${theme.spacings.xhuge};
   }
    
  `}
`;

export const Grid = styled.div`
  /* eslint-disable-next-line no-unused-vars */
  ${({ theme }) => css`
   font-family: ${theme.fonts.family.default};
   counter-reset: grid-counter; 
   display: grid;
   grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
   gap: ${theme.spacings.large};
  `}
`;

export const GridElement = styled.div`
  /* eslint-disable-next-line no-unused-vars */
  ${({ theme }) => css`
   font-family: ${theme.fonts.family.default};
   ${HeadingContainer} {
    position: relative;
    left: 5rem ;
   }

   ${HeadingContainer}::before {
    counter-increment: grid-counter;
    content: counter(grid-counter);
    position: absolute;
    font-size: 7rem;
    top: -3rem;
    left: -5rem;
    transform: rotate(5deg);
   }
    
  `}
`;
