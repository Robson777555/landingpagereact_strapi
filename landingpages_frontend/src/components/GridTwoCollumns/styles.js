import styled, { css } from 'styled-components';
import { Heading } from '../Heading/Heading';

export const Container = styled.div`
  ${({ theme, background, darkMode }) => css`
    display: grid;
    grid-template-columns: 1fr 1fr;
    align-items: center;
    gap: ${theme.spacings.large};
    width: 100%;
    max-width: 1200px;
    margin: 0 auto;
    padding: ${theme.spacings.large};
    background: ${
      background
        ? darkMode
          ? theme.colors.darkBg
          : theme.colors.white
        : 'transparent'
    };
    color: ${darkMode ? theme.colors.white : theme.colors.primaryColor};
    transition: all 300ms ease-in-out;

    @media ${theme.media.lteMedium} {
      grid-template-columns: 1fr;
      text-align: center;
      gap: ${theme.spacings.medium};
    }

    @media ${theme.media.lteSmall} {
      padding: ${theme.spacings.medium} ${theme.spacings.small};
    }

    @media ${theme.media.iphone} {
      padding: ${theme.spacings.small} ${theme.spacings.xsmall};
      gap: ${theme.spacings.small};
    }

    /* Regra específica para 325px de largura e até 852px de altura */
    @media (max-width: 325px) and (max-height: 852px) {
      padding: ${theme.spacings.xsmall} ${theme.spacings.xxsmall || '0.4rem'};
      gap: ${theme.spacings.xsmall};
    }

    ${Heading} {
      color: ${darkMode ? theme.colors.white : theme.colors.black};
      margin-bottom: ${theme.spacings.medium};
      font-size: ${theme.fonts.sizes.xxlarge};
      font-weight: 700;
      text-transform: uppercase;
      line-height: 1.2;
      white-space: nowrap;
      overflow: hidden;
      text-overflow: ellipsis;

      @media ${theme.media.lteMedium} {
        font-size: ${theme.fonts.sizes.xlarge};
        margin-bottom: ${theme.spacings.small};
        white-space: normal;
      }

      @media ${theme.media.lteSmall} {
        font-size: ${theme.fonts.sizes.large};
        margin-bottom: ${theme.spacings.small};
      }

      @media ${theme.media.iphone} {
        font-size: ${theme.fonts.sizes.large};
        margin-bottom: ${theme.spacings.xsmall};
        white-space: nowrap;
        letter-spacing: -0.5px;
        padding: 0 ${theme.spacings.xsmall};
       
      }

      @media (max-width: 325px) and (max-height: 852px) {
        font-size: ${theme.fonts.sizes.medium};
        margin-bottom: ${theme.spacings.xsmall};
        letter-spacing: -0.3px;
        background: red;
      }
    }

    
  `}
`;

export const TextContainer = styled.div`
  ${({ theme }) => css`
    font-family: ${theme.fonts.family.default};
  `};
`;

export const TextComponent = styled.div`
  ${({ theme }) => css`
    font-family: ${theme.fonts.family.default};
  `};
`;

export const ImageContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100%;

    @media ${theme.media.iphone} {
      padding: 0 ${theme.spacings.xsmall};
    }

    @media (max-width: 325px) and (max-height: 852px) {
      padding: 0 ${theme.spacings.xxsmall || '0.4rem'};
    }
  `}
`;

export const Image = styled.img`
  ${({ theme, darkMode }) => css`
    width: 100%;
    max-width: 400px;
    height: auto;
    border-radius: ${theme.spacings.xsmall};
    filter: ${darkMode ? 'brightness(0.85)' : 'none'};
    transition: filter 300ms ease-in-out;

    @media ${theme.media.lteMedium} {
      max-width: 350px;
    }

    @media ${theme.media.lteSmall} {
      max-width: 280px;
    }

    @media ${theme.media.iphone} {
      max-width: 220px;
      margin: 0 auto;
      display: block;
    }

    @media (max-width: 325px) and (max-height: 852px) {
      max-width: 180px;
    }
  `}
`;

export const ToggleButton = styled.button`
  ${({ theme }) => css`
    position: fixed;
    bottom: ${theme.spacings.medium};
    right: ${theme.spacings.medium};
    padding: ${theme.spacings.small} ${theme.spacings.medium};
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    border: none;
    border-radius: 5px;
    cursor: pointer;
    z-index: 1000;
    transition: all 300ms ease-in-out;

    &:hover {
      background: ${theme.colors.secondaryColor};
      transform: scale(1.05);
    }

    @media ${theme.media.iphone} {
      padding: ${theme.spacings.xsmall} ${theme.spacings.small};
      bottom: ${theme.spacings.small};
      right: ${theme.spacings.small};
    }

    @media (max-width: 325px) and (max-height: 852px) {
      padding: ${theme.spacings.xxsmall || '0.4rem'} ${theme.spacings.xsmall};
      bottom: ${theme.spacings.xsmall};
      right: ${theme.spacings.xsmall};
      font-size: ${theme.fonts.sizes.xsmall};
      
    }
  `}
`;
