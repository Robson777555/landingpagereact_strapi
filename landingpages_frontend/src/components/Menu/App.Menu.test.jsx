import { screen, fireEvent } from '@testing-library/react';
import { Menu } from './Menu';
import { renderTheme } from '../../styles/render-theme';
import { describe, it, expect } from '@jest/globals';
import linksMock from '../NavLinks/mock';
const logoData = {
  text: 'Logo',
  link: '#target',
};

describe('<Menu />', () => {
  it('should render logo and Main Menu Nav', () => {
    const { container } = renderTheme(
      <Menu links={linksMock} logoData={logoData} />,
    );
    expect(screen.getByRole('heading', { name: 'Logo' })).toBeInTheDocument();
    expect(
      screen.getByRole('navigation', { name: 'Main menu' }),
    ).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should render menu mobile and button to open and close the menu', () => {
    const { container } = renderTheme(
      <Menu links={linksMock} logoData={logoData} />,
    );
    const button = screen.getByLabelText('Open/Close menu');
    const menuContainer = button.nextSibling;

    expect(button).toBeInTheDocument();
    expect(menuContainer).toBeInTheDocument();
    expect(screen.getByLabelText('Open menu')).toBeInTheDocument();

    fireEvent.click(button);
    expect(screen.getByLabelText('Close menu')).toBeInTheDocument();
    expect(container.querySelector('button')).toBeInTheDocument();
  });

  it('should not render links', () => {
    const { container } = renderTheme(<Menu logoData={logoData} />);
    expect(
      screen.queryByRole('navigation', { name: 'Main menu' }).firstChild,
    ).not.toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
