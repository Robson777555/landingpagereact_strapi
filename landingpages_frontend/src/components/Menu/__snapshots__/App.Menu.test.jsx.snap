// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> should not render links 1`] = `
.c4 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

.c7 {
  color: #0A1128;
  background: transparent;
  padding: 1rem;
  font-size: 2.4rem;
  text-transform: uppercase;
}

.c2 {
  font-family: 'Open Sans',sans-serif;
  position: fixed;
  z-index: 5;
  top: 0;
  left: 0;
  right: 0;
  width: 100%;
  background: #FFFFFF;
  transition: all 300ms ease-in-out;
  border-bottom: 1px solid #eee;
}

.c5 {
  font-family: 'Open Sans',sans-serif;
  max-width: 120rem;
  margin: 0 auto;
  padding: 0 3rem;
  display: flex;
  justify-content: space-between;
  align-items: center;
  background: #FFFFFF;
  height: 10px;
}

.c0 {
  font-family: 'Open Sans',sans-serif;
  z-index: 6;
  position: fixed;
  top: 2rem;
  right: 2rem;
  width: 4rem;
  height: 4rem;
  background: #0A1128;
  color: #FFFFFF;
  border: none;
  border-radius: 50%;
  padding: 0;
  margin: 0;
  display: none;
  align-items: center;
  justify-content: center;
  cursor: pointer;
  transition: transform 0.3s ease;
  pointer-events: all;
}

.c0:hover {
  transform: scale(1.1);
}

.c0 svg {
  width: 2.5rem;
  height: 2.5rem;
  margin: auto;
}

.c8 {
  display: flex;
  align-items: center;
  text-decoration: none;
  color: inherit;
  font-size: 3.2rem;
  font-weight: bold;
  text-transform: uppercase;
}

.c8 >img {
  height: 3rem;
  width: auto;
}

.c9 {
  display: flex;
  flex-flow: row wrap;
  align-items: center;
  justify-content: center;
  gap: 1rem;
}

.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

@media (max-width: 768px) {
  .c2 {
    height: 100vh;
    background: gray;
    display: flex;
    align-items: center;
    justify-content: center;
    overflow: hidden;
  }

  .c2::after {
    content: '';
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background: #FFFFFF;
    opacity: 0;
    visibility: hidden;
    transition: opacity 300ms ease-in-out,visibility 300ms ease-in-out;
    z-index: -1;
  }

  .c2 >.c3 {
    display: none;
    grid-template-columns: 1fr;
    grid-template-rows: 1fr;
    height: 100vh;
    align-items: center;
  }

  .c2 .c6 {
    padding-bottom: 3.2rem;
    display: none;
    justify-content: center;
  }
}

@media (max-width: 768px) {
  .c5 {
    display: flex;
    flex-direction: column;
    text-align: center;
    padding: 2.4rem 0;
    min-height: auto;
    max-height: 90vh;
    overflow-y: auto;
    height: auto;
  }
}

@media (max-width: 768px) {
  .c0 {
    display: flex;
  }
}

@media (max-width: 768px) {
  .c9 {
    flex-flow: column nowrap;
    align-items: stretch;
    width: 100%;
    margin-top: 2.4rem;
    max-height: 80vh;
    overflow-y: auto;
    padding-bottom: 3.2rem;
    gap: 0;
  }
}

<div>
  <button
    aria-label="Open/Close menu"
    class="c0"
  >
    <svg
      aria-hidden="true"
      aria-label="Open menu"
      class="c1"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </button>
  <div
    class="c2"
  >
    <div
      class="c3 c4"
    >
      <div
        class="c5"
      >
        <h1
          class="c6 c7"
        >
          <a
            class="c8"
            href="#target"
            target="_blank"
          >
            Logo
          </a>
        </h1>
        <nav
          aria-label="Main menu"
          class="c9"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<Menu /> should render logo and Main Menu Nav 1`] = `
.c4 {
  max-width: 120rem;
  margin: 0 auto;
  padding: 3.2rem;
}

.c7 {
  color: #0A1128;
  background: transparent;
  padding: 1rem;
  font-size: 2.4rem;
  text-transform: uppercase;
}

.c2 {
  font-family: 'Open Sans',sans-serif;
  position: fixed;
  z-index: 5;
  top: 0;
  left: 0;
  right: 0;
  width: 100%;
  background: #FFFFFF;
  transition: all 300ms ease-in-out;
  border-bottom: 1px solid #eee;
}

.c5 {
  font-family: 'Open Sans',sans-serif;
  max-width: 120rem;
  margin: 0 auto;
  padding: 0 3rem;
  display: flex;
  justify-content: space-between;
  align-items: center;
  background: #FFFFFF;
  height: 10px;
}

.c0 {
  font-family: 'Open Sans',sans-serif;
  z-index: 6;
  position: fixed;
  top: 2rem;
  right: 2rem;
  width: 4rem;
  height: 4rem;
  background: #0A1128;
  color: #FFFFFF;
  border: none;
  border-radius: 50%;
  padding: 0;
  margin: 0;
  display: none;
  align-items: center;
  justify-content: center;
  cursor: pointer;
  transition: transform 0.3s ease;
  pointer-events: all;
}

.c0:hover {
  transform: scale(1.1);
}

.c0 svg {
  width: 2.5rem;
  height: 2.5rem;
  margin: auto;
}

.c8 {
  display: flex;
  align-items: center;
  text-decoration: none;
  color: inherit;
  font-size: 3.2rem;
  font-weight: bold;
  text-transform: uppercase;
}

.c8 >img {
  height: 3rem;
  width: auto;
}

.c10 {
  display: block;
  text-decoration: none;
  font-size: 1.6rem;
  padding: 0.8rem;
  color: #0A1128;
  position: relative;
  text-transform: uppercase;
}

.c10::after {
  content: '';
  position: absolute;
  bottom: 0.5rem;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background-color: #dc143c;
  transition: all 300ms ease-in-out;
}

.c10:hover::after {
  left: 25%;
  width: 50%;
}

.c9 {
  display: flex;
  flex-flow: row wrap;
  align-items: center;
  justify-content: center;
  gap: 1rem;
}

.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

@media (max-width: 768px) {
  .c2 {
    height: 100vh;
    background: gray;
    display: flex;
    align-items: center;
    justify-content: center;
    overflow: hidden;
  }

  .c2::after {
    content: '';
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background: #FFFFFF;
    opacity: 0;
    visibility: hidden;
    transition: opacity 300ms ease-in-out,visibility 300ms ease-in-out;
    z-index: -1;
  }

  .c2 >.c3 {
    display: none;
    grid-template-columns: 1fr;
    grid-template-rows: 1fr;
    height: 100vh;
    align-items: center;
  }

  .c2 .c6 {
    padding-bottom: 3.2rem;
    display: none;
    justify-content: center;
  }
}

@media (max-width: 768px) {
  .c5 {
    display: flex;
    flex-direction: column;
    text-align: center;
    padding: 2.4rem 0;
    min-height: auto;
    max-height: 90vh;
    overflow-y: auto;
    height: auto;
  }
}

@media (max-width: 768px) {
  .c0 {
    display: flex;
  }
}

@media (max-width: 768px) {
  .c10 {
    font-size: 2.4rem;
    padding: 2.4rem;
    text-align: center;
    min-height: 44px;
    display: block;
    width: 100%;
  }

  .c10::after {
    bottom: 0.5rem;
    width: 0;
  }

  .c10:hover::after {
    left: 25%;
    width: 50%;
  }
}

@media (max-width: 768px) {
  .c9 {
    flex-flow: column nowrap;
    align-items: stretch;
    width: 100%;
    margin-top: 2.4rem;
    max-height: 80vh;
    overflow-y: auto;
    padding-bottom: 3.2rem;
    gap: 0;
  }
}

<div>
  <button
    aria-label="Open/Close menu"
    class="c0"
  >
    <svg
      aria-hidden="true"
      aria-label="Open menu"
      class="c1"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </button>
  <div
    class="c2"
  >
    <div
      class="c3 c4"
    >
      <div
        class="c5"
      >
        <h1
          class="c6 c7"
        >
          <a
            class="c8"
            href="#target"
            target="_blank"
          >
            Logo
          </a>
        </h1>
        <nav
          aria-label="Main menu"
          class="c9"
        >
          <a
            class="c10"
            href="#target1"
            rel=""
            target="_self"
          >
            link 1
          </a>
          <a
            class="c10"
            href="#target2"
            rel=""
            target="_self"
          >
            link 2
          </a>
          <a
            class="c10"
            href="#target3"
            rel=""
            target="_self"
          >
            link 3
          </a>
          <a
            class="c10"
            href="#target4"
            rel=""
            target="_self"
          >
            link 4
          </a>
          <a
            class="c10"
            href="#target5"
            rel=""
            target="_self"
          >
            link 5
          </a>
          <a
            class="c10"
            href="#target6"
            rel=""
            target="_self"
          >
            link 6
          </a>
          <a
            class="c10"
            href="#target7"
            rel=""
            target="_self"
          >
            link 7
          </a>
          <a
            class="c10"
            href="#target8"
            rel=""
            target="_self"
          >
            link 8
          </a>
          <a
            class="c10"
            href="#target9"
            rel=""
            target="_self"
          >
            link 9
          </a>
          <a
            class="c10"
            href="#target10"
            rel=""
            target="_self"
          >
            link 10
          </a>
        </nav>
      </div>
    </div>
  </div>
</div>
`;
